block-layer/codecs/dag-cbor.md

```diff
@@ -0,0 +1,41 @@
+# Specification: DagCBOR
+
+**Status: Descriptive - Draft**
+
+DagCBOR supports the full [IPLD Data Model].
+
+CBOR already natively supports all [IPLD Data Model Kinds].
+
+## Format
+
+The CBOR IPLD format is called DagCBOR to disambiguate it from regular CBOR.
+Most CBOR objects are valid DagCBOR. The only hard restriction is that any field
+with the tag 42 must be a valid CID.
+
+## Link Format
+
+As with all IPLD formats, DagCBOR must be able to encode [Links].
+In DagCBOR, links are encoded using the raw-binary identity [Multibase] in a
+field with a byte-string type (major type 2), with the tag 42.
+
+(the inclusion of the Multibase exists for historical reasons)
+
+## Map Key Restriction
+
+In DagCBOR, map keys must be strings.
```

Comment:
Does CBOR even allow non-string keys?

I know we need to add this statement to the data model but I didn’t realize we needed to clear it up here as well.

Reply:
Yes, it does. Also, It would be _really_ nice to support number keys (and I'm pretty sure it won't cause any problems with pathing).

Reply:
We talked about keys in maps in the last sync meeting and thought about restricting the data model to string keys only (for now. Keeping things simple, we can revise that in the future). The issue to track this is https://github.com/ipld/specs/issues/59.